feat: Modified setup.js not to overide "start" script in package.json

Taekyoon opened the pull request:
Added a condition to check if packageJson.scripts["start"] has a value and blocked to override this value. Instead, overriding the "start" value, Added a statement to store this value to "start-hoodie" script in package.json

Taekyoon committed:
feat: Modified setup.js not to overide "start" script in package.json

Added a condition to check if packageJson.scripts["start"] has a value and blocked to override this value. Instead, overriding the "start" value, Added a statement to store this value to "start-hoodie" script in package.json

varjmes commented:
This is for issue #592

varjmes commented on bin/setup.js:
```diff
@@ -15,7 +15,15 @@ if (saveRequested) {
+    log.info('setup', 'start script already set to "' + packageJson.scripts[ 'start' ] +
+    ', you can start hoodie with "npm run start-hoodie" instead')
+
+    packageJson.scripts[ 'start-hoodie' ] = packageJson.script[ 'start' ]
```
Correct me if I am wrong, but:

1. Sometimes you say `packageJson.scripts`, other times you say `packageJson.script`. I think it should always be `packageJson.scripts`?
2. Why are we setting `start-hoodie` to `start`. That feels like the opposite of what we want to do. Shouldn't we be setting `start-hoodie` to `hoodie`?
3. I think there might be an opportunity to write a test for this, somewhere in our unit tests, to prove that this works?

Wondering if @gr2m has any thoughts on my comments above.
Thanks so much for starting this PR ⭐️

gr2m replied:
Correct me if I am wrong, but:

> 1. Sometimes you say `packageJson.scripts`, other times you say `packageJson.script`. I think it should always be `packageJson.scripts`?

👍👁:clap:  great catch! `packageJson.script` should be `packageJson.scripts`

> 2. Why are we setting `start-hoodie` to `start`. That feels like the opposite of what we want to do. Shouldn't we be setting `start-hoodie` to `hoodie`?

Correct. I would suggest to replace line 23 with `packageJson.scripts[ 'start-hoodie' ] = 'hoodie'`

> 3. I think there might be an opportunity to write a test for this, somewhere in our unit tests, to prove that this works?

That’s my fault, I asked @Taekyoon to look into although this part had no tests yet. I’d suggest we test it locally to make sure it works, and add tests in a follow up pull request? But @Taekyoon if you like and feel like it, you can add tests now already, I’m happy to help!

varjmes replied:
That sounds good to me!
Glad I was on the right page here, too :)

Taekyoon replied:
Oh! I think something mistook on the code. I should have written `packageJson.scripts`, instead `packageJson.script`. I'll change this code. And @gr2m, can you let me know how to build `setup.js` test? I've tested in other projects but they already set to run. I tried to find related test codes, but it doesn't exist.

Taekyoon replied:
@gr2m @Charlotteis Now I fixed wrong named variables. And do I need to change line 23 to `packageJson.scripts[ 'start-hoodie' ] = 'hoodie'`? I'm not sure I should change it.

gr2m commented on bin/setup.js:
```diff
@@ -15,7 +15,15 @@ if (saveRequested) {
+
+    packageJson.scripts[ 'start-hoodie' ] = packageJson.script[ 'start' ]
+  } else {
+    packageJson.script[ 'start' ] = 'hoodie'
```
as Charlotte mentioned above, this should be `packageJson.scripts[ 'start' ] = 'hoodie'`

Taekyoon committed:
feat: corrected wrong named variable "packageJson.scripts"

Modified wrong typed variable, changed "packageJson.script -> packageJson.scripts"

gr2m commented on bin/setup.js:
```diff
@@ -15,7 +15,15 @@ if (saveRequested) {
+    log.info('setup', 'start script already set to "' + packageJson.scripts[ 'start' ] +
+    ', you can start hoodie with "npm run start-hoodie" instead')
+
+    packageJson.scripts[ 'start-hoodie' ] = packageJson.scripts[ 'start' ]
```
this should be `packageJson.scripts[ 'start-hoodie' ] = 'hoodie'`, so that we can start the server with `npm run start-hoodie`. The current code sets it to whatever `npm start` does right now, which is not what we want :)

Taekyoon replied:
Got it!

Taekyoon committed:
feat: Changed a statement related to packageJson.scripts[start-hoodie]

Changed value from packageJson.scripts[start] to "hoodie" to input a variable packageJson.scripts[start-hoodie]

gr2m commented:
looks good to me now, thanks for bearing with us @Taekyoon 👍 
Okay for you @Charlotteis to merge now? I’ll help Taekyoon to add tests in a follow up pull request

varjmes commented:
sorry this took so long, just started a new job!

gg team!
